kod/object/passive/spell/creaweap.kod

```diff
@@ -59,14 +60,108 @@ messages:
       return 0;
    }
 
-   CastSpell(who = $,lTargets = $,iSpellPower=$)
+   SomethingSaidDuringTrance(who=$,string=$)
+   {
+      local i, iSpellPower, iCount,isWeaponCreateable, oWeapon, oItemAtt,iSpellPercent;
+
+      iCount = 0;
+      isWeaponCreateable = false;
+
+      iSpellPower = Send(self,@GetSpellPower,#who=who);
+      iSpellPercent = Send(who,@GetSpellAbility,#spell_num=SID_CREATE_WEAPON);
+
+      for i in Send(SYS,@GetItemTemplates)
+      {
+         if(StringEqual(string,Send(i,@GetName)))
+         {
+            if(NOT isClass(i,&Weapon))
+            {
+               return;
+            }
+
+            if(NOT Send(self,@CanCreateWeapon,#sWeapon=Send(i,@GetName),
+                  #iSpellPower=iSpellPower,#iSpellPercent=iSpellPercent))
+            {
+               return;
+            }
+
+            oWeapon = Create(getClass(i));
+            isWeaponCreateable = true;
+            break;
+         }
+      }
+
+      if(isWeaponCreateable = true)
+      {
+         Send(who,@BreakTrance,#event=EVENT_STEER);
+         % Weapon has half its hits, + iSpellPower% of the other half.
+         Send(oWeapon,@SetHits,#number=(Send(oWeapon,@GetHits)*(iSpellPower+100))/200);
+
+         if Send(who,@ReqNewHold,#what=oWeapon)
+         {
+            Send(who,@NewHold,#what=oWeapon);
+            Send(who,@MsgSendUser,#message_rsc=createweapon_cast_rsc,
+                  #parm1=Send(oWeapon,@GetCapIndef),#parm2=Send(oWeapon,@GetName));
+         }
+         else
+         {
+            Send(Send(who,@GetOwner),@NewHold,#what=oWeapon,
+                  #new_row=Send(who,@GetRow),#new_col=Send(who,@GetCol),
+                  #fine_row=(Send(who,@GetFineRow)+2),
+                  #fine_col=(Send(who,@GetFineCol)+2));
+            Send(who,@MsgSendUser,#message_rsc=createweapon_inv_full_rsc,
```

Comment:
> A mace appears in the air before you, but before you can drop something and grab it, it vanishes.

This is a strange message to send when in fact it does not vanish, but falls to your feet. It seems to me like this needs its own unique message or the timer issue needs to get resolved and overweight items "vanished."